nutrition_labels/grant_tagger.py

```diff
@@ -216,20 +179,59 @@ def split_data(self, X_vect, y):
             ]
             # Make sure they are in order otherwise it'll be all 1111s and then 0000s
             sample_index.sort()
-            y = [y[i] for i in sample_index]
-            if self.vectorizer_type == "bert":
-                X_vect = [X_vect[i] for i in sample_index]
-            else:
-                X_vect = X_vect[sample_index]
+
+            # Store data not in sample
+            not_sample_index = data.index.difference(sample_index)
+            unused_data = data.iloc[not_sample_index, :]
+            self.X_unused = unused_data['Grant texts'].tolist()
+            self.y_unused = unused_data[self.label_name].tolist()
+            self.unused_ids = unused_data[train_data_id].tolist()
+            # Sample data
+            data = data.iloc[sample_index, :]
+        else:
+            self.X_unused = []
+            self.y_unused = []
+            self.unused_ids = []
+
+        X = data[["Grant texts", train_data_id]]
+        y = data[self.label_name].tolist()
 
         # Randomly shuffled at the beginning so turn this off
         X_train, X_test, y_train, y_test = train_test_split(
-            X_vect, y, test_size=self.test_size, shuffle=False
-        )
+                X, y, test_size=self.test_size, shuffle=False
+            )
+
+        self.X_train_ids = X_train[train_data_id].tolist()
+        self.X_test_ids = X_test[train_data_id].tolist()
+
+        X_train = X_train['Grant texts'].tolist()
+        X_test = X_test['Grant texts'].tolist()
 
         return X_train, X_test, y_train, y_test
 
     def fit(self, X, y):
+
+        if self.vectorizer_type == "count":
+            self.vectorizer = CountVectorizer(
+                analyzer="word",
+                token_pattern=r"(?u)\b\w+\b",
+                ngram_range=(1, 2),
+                stop_words='english'
+            )
+        elif self.vectorizer_type == "tfidf":
+            self.vectorizer = TfidfVectorizer(
+                analyzer="word",
+                token_pattern=r"(?u)\b\w+\b",
+                ngram_range=(1, 2),
+                stop_words='english'
```

Comment:
I think I might have mentioned that in the past but these params seem set in stone, it would be nice to be able to pass different params from the config.

Reply:
I agree, I'll add this as an issue rather than fix here https://github.com/wellcometrust/nutrition-labels/issues/74